src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -48,19 +48,90 @@ internal class Catalog : SystemTarget, ICatalog, IPlacementContext, ISiloStatusL
         [Serializable]
         internal class DuplicateActivationException : Exception
         {
-            public ActivationAddress ActivationToUse { get; set; }
+            public ActivationAddress ActivationToUse { get; private set; }
 
-            public SiloAddress PrimaryDirectoryForGrain { get; set; } // for diagnostics only!
+            public SiloAddress PrimaryDirectoryForGrain { get; private set; } // for diagnostics only!
+
+            public DuplicateActivationException() : base("DuplicateActivationException") { }
+            public DuplicateActivationException(string msg) : base(msg) { }
+            public DuplicateActivationException(string message, Exception innerException) : base(message, innerException) { }
+
+            public DuplicateActivationException(ActivationAddress activationToUse)
+                : base("DuplicateActivationException")
+            {
+                ActivationToUse = activationToUse;
+            }
+
+            public DuplicateActivationException(ActivationAddress activationToUse, SiloAddress primaryDirectoryForGrain)
+                : base("DuplicateActivationException")
+            {
+                ActivationToUse = activationToUse;
+                PrimaryDirectoryForGrain = primaryDirectoryForGrain;
+            }
+
+            // Implementation of exception serialization with custom properties according to:
+            // http://stackoverflow.com/questions/94488/what-is-the-correct-way-to-make-a-custom-net-exception-serializable
+            protected DuplicateActivationException(SerializationInfo info, StreamingContext context)
+                : base(info, context)
+            {
+                if (info != null)
+                {
+                    ActivationToUse = (ActivationAddress) info.GetValue("ActivationToUse", typeof (ActivationAddress));
+                    PrimaryDirectoryForGrain = (SiloAddress) info.GetValue("PrimaryDirectoryForGrain", typeof (SiloAddress));
+                }
+            }
+
+            public override void GetObjectData(SerializationInfo info, StreamingContext context)
```

Comment:
Validated via unit test:
we do need the extra logic, including the explicit storing in the `SerializationInfo`  and the `GetObjectData`. The article on MSDN https://msdn.microsoft.com/en-us/library/system.exception.aspx stating otherwise appears to be incorrect.